### DISCUSSION TITLE
...Blazor com .NET 8 / ...prática / Criando um Produto

### DISCUSSION THREAD

**Discussion by DrenokBR** [balta-io/forum#506] [Tira Dúvidas] [answered]
Ao final desta aula, quando vou rodar a aplicação minha validação para Preço do Produto não funciona, a mensagem de erro configurada no "Data Annotation Required" para a propriedade "Price" não é disparada quando tento criar um novo produto sem submeter nenhum valor no "input html". Mas a validação para Title do produto funciona normalmente nesse cenário. Ou seja, só funciona a mensagem de erro do Required do Title. Já olhei várias vezes o código e parece estar igual ao do curso.

Será que é porque o C# preenche o tipo Decimal com o valor Zero ?!
Mas no seu código, a validação funciona mesmo tendo o valor Zero no "Input" do Preço. 

Vou colocar as imagens no meu código aqui embaixo.
![ClasseProduct](https://github.com/balta-io/forum/assets/36772075/e2471a7e-d786-4f6a-814c-a4d981aae1d7)

![paginaCreate_Product](https://github.com/balta-io/forum/assets/36772075/88f246b1-2639-4990-9a44-e5b061f153bd)

**Comment by andrebaltieri** [comment, marked as answer]
@DrenokBR boa tarde! Sim é por que seu preço não está como nullable!

**Reply by DrenokBR** [reply]
Fala Balta ! :)

Uma vez que o "decimal" não é do tipo valor, descartei a possibilidade de "setar" a Propriedade Price como Nullable no arquivo "Product.cs", como você ensinou. Então, comparei os códigos das páginas "/Products/Index.razor" e  "/Products/Create.razor" do curso no github e não consegui encontrar onde "setar" o Nullable.

**Reply by DrenokBR** [reply]
Só consegui fazer funcionar usando o atributo "Range" conforme abaixo:

**[Range(1, 10000, ErrorMessage = "Informe o Preço do Produto")]**


![RangeAttribute](https://github.com/balta-io/forum/assets/36772075/5abd8e17-9c43-4bb3-be61-dfaac244f68c)
